cli/src/pcluster/validators/directory_service_validators.py

```diff
@@ -78,3 +78,39 @@ def _validate(self, ldap_tls_reqcert):
                 f"For security reasons it's recommended to use {' or '.join(values_requiring_cert_validation)}",
                 FailureLevel.WARNING,
             )
+
+
+class AdditionalSssdConfigsValidator(Validator):
+    """AdditionalSssdConfigs validator."""
+
+    def _validate(self, additional_sssd_configs, ldap_access_filter):
+        """Validate that AdditionalSssdConfigs does not introduce unacceptable values."""
+        if "ldap_schema" in additional_sssd_configs:
+            value = additional_sssd_configs["ldap_schema"]
+            if value != "AD":
+                self._add_failure(
+                    f"Cannot override the SSSD property 'ldap_schema' "
+                    f"with value '{value}'. "
+                    f"Allowed value is: 'AD'.",
+                    FailureLevel.ERROR,
+                )
+
+        if "id_provider" in additional_sssd_configs:
+            value = additional_sssd_configs["id_provider"]
+            if value != "ldap":
+                self._add_failure(
```

Comment:
the first 2 checks are executing the same code and have the same error in the same form.
What about doing something like:
```python
for config_key, accepted_value in ("ldap_schema", "AD"), ("id_provider", "ldap"):
...
```

Reply:
Or we can also define a RESERVED_AD_SETTINGS dict containing config_key and accepted value in constants.py module, similar to what you added in the cookbook.

Reply:
I agrre in reducing code duplication, but I would put these constants within the validator